File: leetcode/566. Reshape the Matrix/main.ts
// Solved by @kitanoyoru
// https://leetcode.com/problems/reshape-the-matrix/

const matrixReshape = (mat: number[][], r: number, c: number): number[][] => {
  if ((mat.length * mat[0].length) / c != r) {
    return mat
  }

  let ans: number[][] = []
  let row = 0,
    col = 0

  for (let i = 0; i < r; i++) {
    ans.push([])
    for (let j = 0; j < c; j++) {
      ans[i].push(mat[row][col])
      if (col == mat[row].length - 1) {
        row++
        col = 0
      } else {
        col++
      }
    }
  }

  return ans
}

File: leetcode/566. Reshape the Matrix/main.py
# Solved by @kitanoyoru
# https://leetcode.com/problems/reshape-the-matrix/

from typing import List


class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        if (len(mat) * len(mat[0])) / c != r:
            return mat

        ans: List[List[int]] = []

        [row, col] = [0, 0]

        for i in range(r):
            ans.append([])
            for _ in range(c):
                ans[i].append(mat[row][col])
                if col == len(mat[row]) - 1:
                    row += 1
                    col = 0
                else:
                    col += 1

        return ans
